### DISCUSSION TITLE
Need of OpenSSL thread support to work with providers.

### DISCUSSION THREAD

**Discussion by jsrikanth1608** [openssl/openssl#24905] [Q&A] [answered]
Hi,

My application is single threaded and want to use multiple providers (SW and HW). If I configure OpenSSL 3.2.x library with "no-threads" option, will there be any consequences at run time for my use case..?  Please let us know, is it mandatory to enable threads support in the OpenSSL to work with single or multiple providers..

**Comment by nhorman** [comment, marked as answer]
The use of multiple providers is orthogonal to the use of multiple threads.  You should be able to configure openssl with no-threads, and still use several providers.  The usual threading rules apply however - i.e. if you have a 3rd party provider that creates a thread that calls back into openssl, thats going to produce unpredictable results

**Comment by jsrikanth1608** [comment]
Thank you @nhorman 
In case multiple threads using same or different providers we need to configure thread support in OpenSSL, is it correct..?

**Reply by nhorman** [reply]
correct

**Reply by jsrikanth1608** [reply]
Thank you @nhorman
